app/src/main/java/ch/epfl/sdp/blindly/main_screen/MatchPageFragment.kt

```diff
@@ -34,7 +40,37 @@ class MatchPageFragment : Fragment() {
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
         // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_match_page, container, false)
+        val view = inflater.inflate(R.layout.fragment_match_page, container, false)
+
+        val intent = Intent(context, MatchActivity::class.java)
+        val profiles = createProfiles()
+        intent.putParcelableArrayListExtra("EXTRA_MATCH_PROFILES", profiles as ArrayList<out Parcelable>)
+        val match_activity_button = view.findViewById<Button>(R.id.button_to_match)
+        match_activity_button.setOnClickListener{startActivity(intent)}
```

Comment:
Might be a bit picky but usually in the project we used the camelCase for the val and var, I think it would be better to be consistent and use matchActivityButton for the name